include/ada/checkers-inl.h

```diff
@@ -13,49 +13,57 @@
 
 namespace ada::checkers {
 
-inline bool has_hex_prefix_unsafe(std::string_view input) {
+inline bool has_hex_prefix_unsafe(std::string_view const input) {
   // This is actually efficient code, see has_hex_prefix for the assembly.
   uint32_t value_one = 1;
-  bool is_little_endian = (reinterpret_cast<char*>(&value_one)[0] == 1);
+  bool const is_little_endian = (reinterpret_cast<char*>(&value_one)[0] == 1);
   uint16_t word0x{};
-  std::memcpy(&word0x, "0x", 2);  // we would use bit_cast in C++20 and the
-                                  // function could be constexpr.
+
+  static constexpr auto zero_x = "0x";
+  // we would use bit_cast in C++20 and the function could be constexpr.
+  std::memcpy(&word0x, static_cast<void const*>(zero_x), 2);
+
   uint16_t two_first_bytes{};
   std::memcpy(&two_first_bytes, input.data(), 2);
   if (is_little_endian) {
-    two_first_bytes |= 0x2000;
+    two_first_bytes |= 0x2000U;
   } else {
-    two_first_bytes |= 0x020;
+    two_first_bytes |= 0x020U;
```

Comment:
You are replacing `0x020` with `0x020U`. Why is this needed? Can you explain when the new code would give different results?

Reply:
Signed integers are not recommended to be used in bitwise operations.